server/app.js

```diff
@@ -420,6 +420,14 @@ if(isLocalEnvironment){
 	});
 }
 
+// Catch-all route for invalid routes
+app.use((req, res, next) => {
+	if (!req.route)
+		return res.redirect('/');
+
+	return next();
+});
+

```

Comment:
Can this be merged into a check in the other function in line 462?

Reply:
The rerouting needs to happen before rendering the page; otherwise we spend resources rendering a blank page that never gets sent.

Reply:
Right, so:

```
app.use(asyncHandler(async (req, res, next)=>{
	if (!req.route)
		return res.redirect('/');

        const page = await renderPage(req, res);
	if(!page) return;
	res.send(page);
});
```

Or am i missing something? ( i haven't tested if this would work)

Reply:
Rendering the page happens starting at line 432. Rerouting needs to happen before that. Line 462 is too late.

Reply:
Ah wait, 432 is just defining the render function, but it doesn't get called until 462. I suppose we could merge them, but I would then want to move the `renderPage()` definition down. I.e., I want all "route/reroute" handling to be together without `renderPage()` definition in between